tests/unit/test_lambda.py

```diff
@@ -89,6 +90,128 @@ def test_delete_event_source_mapping(self):
             self.assertEqual(json.loads(result.get_data()).get('UUID'), self.TEST_UUID)
             self.assertEqual(0, len(lambda_api.event_source_mappings))
 
+    def test_invoke_RETURNS_415_WHEN_not_json_input(self):
+        with self.app.test_request_context() as context:
+            context.request._cached_data = '~notjsonrequest~'
+            response = lambda_api.invoke_function(self.FUNCTION_NAME)
+            self.assertEqual('415 UNSUPPORTED MEDIA TYPE', response.status)
+
+    def _request_response(self, context):
+        context.request._cached_data = '{}'
+        context.request.args = {'Qualifier': '$LATEST'}
+        context.request.environ['HTTP_X_AMZ_INVOCATION_TYPE'] = 'RequestResponse'
+        self._create_function(self.FUNCTION_NAME)
+
+    @mock.patch('localstack.services.awslambda.lambda_api.run_lambda')
+    def test_invoke_plain_text_response(self, mock_run_lambda):
+        with self.app.test_request_context() as context:
+            self._request_response(context)
+            mock_run_lambda.return_value = '~notjsonresponse~'
+            response = lambda_api.invoke_function(self.FUNCTION_NAME)
+            self.assertEqual('~notjsonresponse~', response[0])
+            self.assertEqual(200, response[1])
+            self.assertEqual({'Content-Type': 'text/plain'}, response[2])
+
+    @mock.patch('localstack.services.awslambda.lambda_api.run_lambda')
```

Comment:
nitpick: We could probably reduce duplication between lines 115-213 if we create a small util function that gets called from the different test methods, passing in the mock `return_value` and the expected output.

Reply:
agreed

Reply:
No worries at all, would be great if we can pick this up in a follow-up PR. Going to merge the changes now. Thanks a bunch!